Monitor Query UX Study Feedback (#19771)
Azure/azure-sdk-for-go: sdk/monitor/azquery/README.md

The Azure Monitor Query client library is used to execute read-only queries against [Azure Monitor][azure_monitor_overview]'s two data platforms:

> - [Logs](https://docs.microsoft.com/azure/azure-monitor/logs/data-platform-logs) - Collects and organizes log and performance data from monitored resources. Data from different sources such as platform logs from Azure services, log and performance data from virtual machines agents, and usage and performance data from apps can be consolidated into a single [Azure Log Analytics workspace](https://docs.microsoft.com/azure/azure-monitor/logs/data-platform-logs#log-analytics-and-workspaces). The various data types can be analyzed together using the [Kusto Query Language][kusto_query_language]. See the [Kusto to SQL cheat sheet][kusto_to_sql] for more information.

Author: fix in later PRs

> - [Metrics](https://docs.microsoft.com/azure/azure-monitor/essentials/data-platform-metrics) - Collects numeric data from monitored resources into a time series database. Metrics are numerical values that are collected at regular intervals and describe some aspect of a system at a particular time. Metrics are lightweight and capable of supporting near real-time scenarios, making them particularly useful for alerting and fast detection of issues.

Author: fixed in later PRs

### Authentication

> This document demonstrates using [azidentity.NewDefaultAzureCredential][default_cred_ref] to authenticate. Client accepts any [azidentity][azure_identity] credential. See the [azidentity][azure_identity] documentation for more information about other credential types.

Reviewer: ```suggestion
This document demonstrates using [azidentity.NewDefaultAzureCredential][default_cred_ref] to authenticate. The client accepts any [azidentity][azure_identity] credential. See the [azidentity][azure_identity] documentation for more information about other credential types.
```

### Timespan

> It's best practice to always query with a timespan (type ISO8601TimeInterval) to prevent excessive queries of the entire logs or metrics data set. Logs uses the ISO8601 Time Interval Standard. All time should be in UTC.

Reviewer: ```suggestion
It's best practice to always query with a timespan (type `ISO8601TimeInterval`) to prevent excessive queries of the entire logs or metrics data set. Logs uses the ISO8601 Time Interval Standard. All time should be represented in UTC.
```

> Use the NewISO8601TimeInterval() method for easy creation.

Reviewer: ```suggestion
Use the `NewISO8601TimeInterval()` method for easy creation.
```

		|---Name *string
		|---Type *LogsColumnType
	|---Name *string
	|---Rows []Row               // Rows contain the actual results of the query
|---Error *ErrorInfo
	|---Code *string
|---Visualization []byte
|---Statistics []byte
```

### Batch query

> `QueryBatch` is an advanced method allowing users to execute multiple logs queries in a single request. It takes in a [BatchRequest](#batch-query-request-structure) and returns a [BatchResponse](#batch-query-result-structure). `QueryBatch` can return results in any order (usually in order of completion/success). Please use the `CorrelationID` field to identify the correct response. 

Reviewer: ```suggestion
`QueryBatch` is an advanced method allowing users to execute multiple logs queries in a single request. The method accepts a [BatchRequest](#batch-query-request-structure) and returns a [BatchResponse](#batch-query-result-structure). `QueryBatch` can return results in any order (usually in order of completion/success). Use the `CorrelationID` field to identify the correct response. 
```

> To get visualization data for logs queries, set `include-render` to true in LogsClientQueryWorkspaceOptions Prefer string.

Reviewer: ```suggestion
To get visualization data for logs queries, set `include-render` to true in LogsClientQueryWorkspaceOptions Prefer string.
```
```suggestion
To get visualization data for logs queries, set `include-render` to `true` in the `LogsClientQueryWorkspaceOptions` `Prefer` string.
```

```go
azquery.LogsClientQueryWorkspaceOptions{Prefer: to.Ptr("wait=600,include-statistics=true,include-render=true")}
```

Example QueryWorkspace options: [link][example_queryworkspace_2]

> To do the same with `QueryBatch` set the values in the `BatchQueryRequest.Headers` map with a key of "prefer".

Reviewer: ```suggestion
To do the same with `QueryBatch`, set the values in the `BatchQueryRequest.Headers` map with a key of "prefer".
```

[azure_monitor_overview]: https://docs.microsoft.com/azure/azure-monitor/overview
[context]: https://pkg.go.dev/context
[cloud_documentation]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud

> [create_workspace]: https://learn.microsoft.com/en-us/azure/azure-monitor/logs/quick-create-workspace?tabs=azure-portal

Reviewer: Omit the locale string from docs URLs
```suggestion
[create_workspace]: https://learn.microsoft.com/azure/azure-monitor/logs/quick-create-workspace
```

[example_query_workspace]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery#example-LogsClient.QueryWorkspace
[example_queryworkspace_2]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery#example-LogsClient.QueryWorkspace-Second
[example_logs_client]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery#NewLogsClient
[example_metrics_client]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery#NewMetricsClient
[example_metrics_listdefinitions]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery#example-MetricsClient.NewListDefinitionsPager
[example_metrics_listnamespaces]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery#example-MetricsClient.NewListNamespacesPager
[example_metrics_queryresource]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery#example-MetricsClient.QueryResource
[example_timespan]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery#NewISO8601TimeInterval
[kusto_query_language]: https://learn.microsoft.com/azure/data-explorer/kusto/query/

> [kusto_to_sql]: https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/sqlcheatsheet

Reviewer: ```suggestion
[kusto_to_sql]: https://learn.microsoft.com/azure/data-explorer/kusto/query/sqlcheatsheet
```
